### PR TITLE
Add Smith (Smedjen) building template

### PR DESCRIPTION
**Pull request by nailorcngci**
smith-ss

### CODE REVIEW

File: assets/blocks/brownCornerRoof.block [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "displayName": "Brown Corner Roof",
+    "family": "horizontal",
+    "shape": "engine:slopecorner",
```
Comment (skaldarnar) [inline assets/blocks/brownCornerRoof.block:4]: I don't think it is necessary to declare own block types for `brownCornerRoof` and `brownRoof` (we'll need the texture, though). 
In-game, you can retrieve the equivalent blocks via

```
give hjerlhede:brownroof 16 engine:slopecorner
give hjerlhede:brownroof 16 engine:slope
```

File: assets/prefabs/structures/Smith.prefab [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "StructureTemplate": {
+    },
+    "SpawnBlockRegions": {
+        "regionsToFill": [
+            { "blockType": "CoreBlocks:DoorBottom.BACK", "region": { "min": [-2, 1, 1], "size": [2, 1, 1]}},
+            { "blockType": "CoreBlocks:DoorTop.BACK", "region": { "min": [-2, 2, 1], "size": [2, 1, 1]}},
+            { "blockType": "CoreBlocks:Glass", "region": { "min": [-4, 2, 3], "size": [1, 1, 2]}},
+            { "blockType": "CoreBlocks:Glass", "region": { "min": [4, 2, 3], "size": [1, 1, 2]}},
+            { "blockType": "CoreBlocks:Glass", "region": { "min": [2, 2, 6], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.BACK", "region": { "min": [-2, 6, 4], "size": [1, 1, 1]}},
```
Comment (skaldarnar) [inline assets/prefabs/structures/Smith.prefab:11]: Given my comment above, all occurrences of `HjerlHede:brownCornerRoof` can be replaced by `hjerlhede:brownroof:engine:slopecorner` in the structural template prefab.

File: assets/prefabs/structures/Smith.prefab [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            { "blockType": "HjerlHede:brownCornerRoof.FRONT", "region": { "min": [4, 4, 1], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.FRONT", "region": { "min": [3, 5, 2], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.FRONT", "region": { "min": [2, 6, 3], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.LEFT", "region": { "min": [-4, 4, 1], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.LEFT", "region": { "min": [-3, 5, 2], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.LEFT", "region": { "min": [-2, 6, 3], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.RIGHT", "region": { "min": [2, 6, 4], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.RIGHT", "region": { "min": [3, 5, 5], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownCornerRoof.RIGHT", "region": { "min": [4, 4, 6], "size": [1, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.BACK", "region": { "min": [-1, 6, 4], "size": [3, 1, 1]}},
```
Comment (skaldarnar) [inline assets/prefabs/structures/Smith.prefab:23]: Equivalently, all occurrences of `HjerlHede:brownRoof` can be replaced by `hjerlhede:brownroof:engine:slope` as mentioned above.

File: assets/prefabs/structures/Smith.prefab [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            { "blockType": "HjerlHede:brownRoof.BACK", "region": { "min": [-2, 5, 5], "size": [5, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.BACK", "region": { "min": [-3, 4, 6], "size": [7, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.FRONT", "region": { "min": [-3, 4, 1], "size": [7, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.FRONT", "region": { "min": [-2, 5, 2], "size": [5, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.FRONT", "region": { "min": [-1, 6, 3], "size": [3, 1, 1]}},
+            { "blockType": "HjerlHede:brownRoof.LEFT", "region": { "min": [-4, 4, 2], "size": [1, 1, 4]}},
+            { "blockType": "HjerlHede:brownRoof.LEFT", "region": { "min": [-3, 5, 3], "size": [1, 1, 2]}},
+            { "blockType": "HjerlHede:brownRoof.RIGHT", "region": { "min": [4, 4, 2], "size": [1, 1, 4]}},
+            { "blockType": "HjerlHede:brownRoof.RIGHT", "region": { "min": [3, 5, 3], "size": [1, 1, 2]}},
+            { "blockType": "HjerlHede:sorringladendiagonal", "region": { "min": [-3, 1, 1], "size": [1, 1, 1]}},
```
Comment (skaldarnar) [inline assets/prefabs/structures/Smith.prefab:33]: I'm wondering how/why you use both `HjerlHede:sorringladendiagonal` and `StructuralResources:TimberFrameDiagonal`. Is this by mistake or on purpose? 

I would appreciate if you could use resources from StructuralResources where possible, i.e., replace the occurrences of the `sorringladen` blocks with the equivalents from StructuralResources.

**Review by skaldarnar — CHANGES_REQUESTED** [review]
Great contribution overall! Thank you very much! 👍 

I left a few comments regarding the blocks you've used and whether this PR should introduce new blocks or not. I'm fine with merging it as is, but would highly appreciate if you could do some of the changes (or comment on them) 😉

File: assets/blocks/brownCornerRoof.block (hunk 1 above)
Comment (nailorcngci) [reply assets/blocks/brownCornerRoof.block:4]: Both giving brownRoof somehow.

File: assets/prefabs/structures/Smith.prefab (hunk 3 above)
Comment (nailorcngci) [reply assets/prefabs/structures/Smith.prefab:23]: I tried that and command above. It didn't work out it is not placing any blocks.
ss111

File: assets/prefabs/structures/Smith.prefab (hunk 4 above)
Comment (nailorcngci) [reply assets/prefabs/structures/Smith.prefab:33]: `HjerlHede:sorringladendiagonal` and `StructuralResources:TimberFrameDiagonal` have different directions one of them is to the left and other is to the right.

 I changed `HjerlHede:sorringladenplain` to `StructuralResources:TimberFramePlain` and `HjerlHede:sorringladenmiddle` to `StructuralResources:TimberFrameVertical`.

File: assets/prefabs/structures/Smith.prefab (hunk 3 above)
Comment (skaldarnar) [reply assets/prefabs/structures/Smith.prefab:23]: Hm, that's weird. 🤔 Let's merge it in the current state, we can still improve this later on. 😉

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**